### PR TITLE
Test to simulate a checkout workflow.

### PR DESCRIPTION
**Pull request by praveenkumarct**
- Added test setup for payment transactions workflow.
- Test to Verify credit card payone payment transaction.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -75,6 +75,7 @@ subprojects {
         implementation "ch.qos.logback:logback-access:${depVersions.logbackClassic}"
         implementation "org.slf4j:jul-to-slf4j:${depVersions.julToSlf4j}"
         implementation "org.apache.httpcomponents:httpclient:${depVersions.apacheHttp}"
+        implementation "com.google.guava:guava:30.1.1-jre"
```
Comment (lojzatran) [inline build.gradle:78]: I thought we removed this? Why did you add it back?

File: service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java [hunk 2]
```diff
@@ -43,11 +62,11 @@ public void setUp() throws URISyntaxException {
...
-    // to ensure that payone-integration service will work as expected
     @Test
-    public void makeBankTransferPayoneRequest() throws IOException {
-        HttpResponse httpResponse = executeGetRequest(format(URL_PATTERN, 8080));
-        assertThat(httpResponse.getStatusLine().getStatusCode()).isEqualTo(200);
+    public void verifyCreditCardTransferPayoneRequestIsSuccess() throws Exception {
+        final String paymentId = createPayment("CREDIT_CARD", "CHARGE", "40", "EUR");
+
+        HttpResponse httpResponse = executeGetRequest(format(URL_HANDLE_PAYMENT+paymentId, DEFAULT_PORT));
+        assertThat(httpResponse.getStatusLine().getStatusCode()).isIn(HttpStatus.SC_ACCEPTED, HttpStatus.SC_OK);
```
Comment (lojzatran) [inline service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java:70]: Maybe you can assert more fields in the payment? Just fetch the payment object by ID and assert?

File: service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java [hunk 3]
```diff
@@ -16,25 +16,44 @@
...
-        testInternalProperties.put("TENANTS", "TEST_DATA");
-        testInternalProperties.put("TEST_DATA_CT_PROJECT_KEY", CT_PROJECT_KEY);
-        testInternalProperties.put("TEST_DATA_CT_CLIENT_ID", CT_CLIENT_ID);
-        testInternalProperties.put("TEST_DATA_CT_CLIENT_SECRET", CT_CLIENT_SECRET);
-        testInternalProperties.put("TEST_DATA_PAYONE_KEY", "TvIOwFdSzOSVKE3Y");
-        testInternalProperties.put("TEST_DATA_PAYONE_MERCHANT_ID", "31102");
-        testInternalProperties.put("TEST_DATA_PAYONE_PORTAL_ID", "2022125");
-        testInternalProperties.put("TEST_DATA_PAYONE_SUBACC_ID", "31281");
+        cleanupData();
+        testInternalProperties.put("TENANTS", getTenant());
```
Comment (lojzatran) [inline service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java:39]: Please also update the documentation how to run this project for devs so next time a new dev doesn't have to ask for info.

File: service/src/test/java/util/PaymentRequestHelperUtil.java [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+import static util.PropertiesHelperUtil.getPayoneMerchantId;
+import static util.PropertiesHelperUtil.getPayonePortalId;
+import static util.PropertiesHelperUtil.getPayoneSubAccId;
+import static util.PropertiesHelperUtil.getProjectKey;
+import static util.PropertiesHelperUtil.getTenant;
+
+public class PaymentRequestHelperUtil {
+
+    public static final Duration CTP_REQUEST_TIMEOUT = Duration.ofMinutes(2);
+    public static String PSEUDO_CARD_PAN = null;
```
Comment (lojzatran) [inline service/src/test/java/util/PaymentRequestHelperUtil.java:70]:
```suggestion
    public static String pseudoCardPan = null;
```

File: service/src/test/java/util/PaymentRequestHelperUtil.java [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+import static util.PropertiesHelperUtil.getPayoneKey;
+import static util.PropertiesHelperUtil.getPayoneMerchantId;
+import static util.PropertiesHelperUtil.getPayonePortalId;
+import static util.PropertiesHelperUtil.getPayoneSubAccId;
+import static util.PropertiesHelperUtil.getProjectKey;
+import static util.PropertiesHelperUtil.getTenant;
+
+public class PaymentRequestHelperUtil {
+
+    public static final Duration CTP_REQUEST_TIMEOUT = Duration.ofMinutes(2);
```
Comment (lojzatran) [inline service/src/test/java/util/PaymentRequestHelperUtil.java:69]: Do you need to make all of them public?

File: service/src/test/java/util/PaymentRequestHelperUtil.java [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+        Matcher m = p.matcher(cardPanResponse);
+
+        if (!m.matches()) {
+            throw new RuntimeException(format("Unexpected pseudocardpan response: %s", cardPanResponse));
+        }
+
+        return m.group(1);
+    }
+
+    public static void cleanupData() {
```
Comment (lojzatran) [inline service/src/test/java/util/PaymentRequestHelperUtil.java:229]: We discussed yesterday that the better approach is not to clean up data, but make the tests in a way that they will run no matter what is in the project. Can you remove the clean up so we test this approach with this project?

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (praveenkumarct) [reply build.gradle:78]: I am using `com.google.common.collect.ImmutableList` from this library to generate the request object. So I added it.
check [here](https://github.com/commercetools/commercetools-payone-integration/pull/392/files#diff-5908522bce9ab5c456c3d4f15a715996870392cc55574720e636b1f6c0ddcdfcR198)

Comment (lojzatran) [reply build.gradle:78]: Can you replace it with normal JDK list and remove guava? I don't think it's a good idea to add a library only for this purpose.

File: service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java (hunk 2 above)
Comment (praveenkumarct) [reply service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java:80]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/f04cec2a5e760744ac66b6102ca247dc52629d1c

File: service/src/test/java/util/PaymentRequestHelperUtil.java (hunk 4 above)
Comment (praveenkumarct) [reply service/src/test/java/util/PaymentRequestHelperUtil.java]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/448b682057ccc97f833fb049f53f918266fb4f08

File: service/src/test/java/util/PaymentRequestHelperUtil.java (hunk 5 above)
Comment (praveenkumarct) [reply service/src/test/java/util/PaymentRequestHelperUtil.java]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/530108314fae7a172c783b70db7bc7b6d82417d6

File: service/src/test/java/util/PaymentRequestHelperUtil.java (hunk 6 above)
Comment (praveenkumarct) [reply service/src/test/java/util/PaymentRequestHelperUtil.java]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/b39674f48631147fc0916ee71ec7bb496d960ff4

File: service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java (hunk 3 above)
Comment (praveenkumarct) [reply service/src/test/java/com/commercetools/main/BasicPaymentRequestWorkflow.java:44]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/b2aae19711c4b8ef87c605edee260d8a61dff561

File: build.gradle (hunk 1 above)
Comment (praveenkumarct) [reply build.gradle:78]: Not really only for List. I am using guava library for few other things. Example `Preconditions` , creation on immutable map and immutable list. 
Do you see any issue by adding this library?

Comment (lojzatran) [reply build.gradle:78]: Yes, there were issues with this library and we could not properly update it. That's why we removed it recently. See here: https://github.com/commercetools/commercetools-payone-integration/pull/370. And we're not alone with these problems: https://www.reddit.com/r/java/comments/d3gi2g/do_you_still_use_guave_and_if_you_dont_anymore/. So avoid it if you can.

Comment (praveenkumarct) [reply build.gradle]: https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/afaab776a051a57299e262fd3f59683fea7f5a08

**Review by lojzatran — APPROVED** [review]
